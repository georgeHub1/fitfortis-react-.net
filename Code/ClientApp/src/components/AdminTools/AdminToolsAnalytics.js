import React, { Component } from 'react';
import { Tag, Statistic, Form } from 'antd';
import { injectIntl } from 'react-intl';
import { connect } from 'react-redux';
import ReactLoading from 'react-loading';
import { bindActionCreators } from 'redux';
import ChartAreaAnalyticsSparkline from '../Common/ChartAreaAnalyticsSparkline';
import { fetchAnalyticsDetails, getAnalyticDetails } from '../../redux/adminTool.action';
import { setMetricsDateRange } from '../../redux/metricsDateRange.action';
import CustomCard from '../Common/CustomCard.js';
import * as R from 'ramda';
import analyticId from '../../utils/AnalyticsAndAbTests.tsx';

class AdminToolsAnalytics extends Component {
  displayName = AdminToolsAnalytics.name
  constructor () {
    super();
    this.state = {};
  }

  componentDidMount () {
    analyticId.firebaseAnalyticsLog('AdminTools_Analytics');
  }
  componentWillMount () {
    const { intl: { locale } } = this.props;

    this.props.fetchAnalyticsDetails(locale);
    this.props.fetchAnalyticsUserDEtail(locale);
    this.props.setMetricsDateRange(this.dateRange());
  }

  dateRange = () => {
    const currentTime = new Date().getTime();

    return { from: currentTime - 2592000000, to: currentTime + 172800000, key: 'YEAR_KEY' };
  };

  render () {
    const { adminTool: { isAnalyticFetchingData, analyticData, analyticChartDetails, isAnalyticChartDetailsLoading }, metricsDateRange } = this.props;
    const data = analyticData !== null
    ? R.groupWith((a, b) => a.category === b.category, R.sort(R.ascend(R.prop('category')), R.clone(analyticData)))
    : [];

    return (
      <div className="AdminToolsAnalyticsSection">
        {
          (!isAnalyticFetchingData && !isAnalyticChartDetailsLoading)
          ? <div className="AdminToolsAnalyticsLeft">
            {
              data.map((items, index) => {
                return (
                  <CustomCard key={index}>
                    <div>
                      <div className="cardHeaderProfile">
                        { items[0].category }
                      </div>
                      <div className="containerTags adminToolTags">
                        <Form.Item>
                          {
                            R.groupWith((a, b) => a.name === b.name, R.sort(R.ascend(R.prop('name')), R.clone(items))).map((item, itemIndex) => {
                              return (
                                <Tag key={itemIndex} className='uncheckedTag'>
                                  {analyticChartDetails.find(x => x.analytic.id === item[0].analyticData.analyticId).analyticData.length === 0
                                    && <Statistic title={item[0].name} value={item[0].analyticData.value} />
                                  }
                                  {
                                    analyticChartDetails.find(x => x.analytic.id === item[0].analyticData.analyticId).analyticData.length > 0
                                    && <ChartAreaAnalyticsSparkline
                                      latestValue={item[0].analyticData.value}
                                      metricKey={item[0].analyticData.id}
                                      data={analyticChartDetails.find(x => x.analytic.id === item[0].analyticData.analyticId).analyticData}
                                      metricName={item[0].name}
                                      metricsDateRange={metricsDateRange}
                                    />
                                  }
                                </Tag>
                              );
                            })
                          }
                        </Form.Item>
                      </div>
                    </div>
                  </CustomCard>
                );
              })
            }
          </div>
          : <ReactLoading className="react-loading-center" type="spin" color="#335889"  />
        }
      </div>
    );
  }
}

const mapStateToProps = (({ adminTool, metricsDateRange }) => ({ adminTool, metricsDateRange }));

const mapDispatchToProps = dispatch => ({
  fetchAnalyticsDetails: bindActionCreators(fetchAnalyticsDetails, dispatch),
  fetchAnalyticsUserDEtail: bindActionCreators(getAnalyticDetails, dispatch),
  setMetricsDateRange: bindActionCreators(setMetricsDateRange, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(injectIntl(AdminToolsAnalytics));

