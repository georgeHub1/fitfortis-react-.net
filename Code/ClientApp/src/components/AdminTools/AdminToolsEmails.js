import React from 'react';
import { Alert } from 'antd';
import { FormattedMessage } from 'react-intl';
import { sorting } from './const';
import styles from './styles.module.less';

const AdminToolsEmails = ({activeEmail, adminUserList}) => {
  const Emails = ({activeEmail}) => {
    return sorting(adminUserList).map(data => (
      <span className={`${styles.email_list} ${(activeEmail === data.email) ? styles.active : ''}` } key={data.email}>{data.email}</span>
    ));
  };

  return (
    <div className={styles.adminToolEmail}>
      <Alert
        type="error"
        message={<FormattedMessage id='AdminToolsEmails.adminAccounts' />}
        description={<Emails activeEmail={activeEmail}/>}
      />
    </div>
  );
};

export default AdminToolsEmails;
