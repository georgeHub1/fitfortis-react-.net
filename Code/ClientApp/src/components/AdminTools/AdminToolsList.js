import React from 'react';
import { getCountryIcon, getValidFormat } from './utils';
import { AdminToolsViewMobile } from './AdminToolsViewMobile';
import { Table } from 'antd';

const AdminToolsList = ({filteredData, isMobile, columns, getTrimText, MenuItems, visible, onVisibleChange}) => {
  return (
    !isMobile
    ? (
        <div>
          <div className="userDetailsTable">
            {
              <Table
                rowKey={record => record.id}
                columns={columns}
                dataSource={filteredData}
                pagination={false}
              />
            }
          </div>
        </div>
      ) : (
      <div>
      {
        <div className="adminToolsdocumentTabs">
          {
            filteredData.map((data, index) => {
              data = getValidFormat(data);
              return (
                <AdminToolsViewMobile
                    data={data}
                    visible={visible}
                    onVisibleChange={onVisibleChange}
                    title={data.title}
                    description={getTrimText(data.description, true)}
                    date={data.date}
                    language={getCountryIcon(data.language)}
                    id={data.id}
                    key={index}
                    menu={<MenuItems data={data} />}
                />
              );
            })
          }
        </div>
      }
      </div>
    )
  );
};

export default AdminToolsList;
