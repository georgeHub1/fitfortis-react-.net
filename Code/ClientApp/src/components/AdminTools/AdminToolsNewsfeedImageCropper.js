import React from 'react';
import { Modal, Button, Icon } from 'antd';
import PropTypes from 'prop-types';
import AvatarEditor from 'react-avatar-editor';
import styles from './styles.module.less';

class AdminToolsNewsfeedImageCropper extends React.Component {
  constructor (props) {
    super(props);
    this.state = {
			scale: 1.2,
			rotate: 0,
			allowZoomOut: false,
			borderRadius: 0,
			position: { x: 0.5, y: 0.5 },
			preview: null,
			width: 1440,
			height: 900
    };
  }

  reset = () => {
		this.setState({
      base64Img: '',
      croppedImageUrl: ''
    });
	}
  rotateLeft = e => {
    e.preventDefault();
    this.setState({
      rotate: this.state.rotate - 90
    });
  }
  rotateRight = e => {
      e.preventDefault();
      this.setState({
          rotate: this.state.rotate + 90
      });
  }
	setEditorRef = editor => {
		if (editor) this.editor = editor;
	};
	onLoadSuccess = info => {
		this.rotateImage();
  }

	rotateImage () {
    const img = this.editor.getImageScaledToCanvas().toDataURL();

		this.setState({ croppedImageUrl: img, base64Img: img });
	}
	onImageChange = () => {
		this.rotateImage();
	}

  componentDidMount () {
    this.scaleSize();
  }
  scaleSize () {
    const {avatar} = this.props;
    const img = new Image();

    img.src = avatar;
    img.onload = () => {
      if (img.width === 1440 && img.height === 900) {
        this.setState({scale: 1});
      }
    };
  }
  render () {
    const { avatar, visible, handleCancel, handleOk, formatMessage } = this.props;
    const { base64Img, rotate, croppedImageUrl, scale, height, width, borderRadius } = this.state;

    return (
      <Modal
        width={600}
        closable={false}
        className="wrapperModal"
        title={formatMessage({id: 'AdminToolsNewsfeedImageCropper.cropHeader'})}
        visible={visible}
        onOk={() => {
          handleOk(base64Img);
          this.reset();
        }}
        onCancel={() => {
          handleCancel();
          this.reset();
        }}
      >
        <div className={`${styles.cropWrapper} adminToolCropper`}>
					<AvatarEditor
						ref={this.setEditorRef}
						image={avatar}
						width={width}
						height={height}
						color={[255, 255, 255, 0.6]} // RGBA
						scale={scale}
						border={50}
						borderRadius={borderRadius}
						onLoadSuccess={this.onLoadSuccess}
						onImageChange={this.onImageChange}
            rotate={rotate}
            style={{
              width: '100%',
              height: 'auto'
            }}
					/>
        </div>
				<div className={styles.croppedImageRotate}>
        <Button onClick={this.rotateLeft}><Icon type="redo" rotate={90} /></Button>
          <Button onClick={this.rotateRight}><Icon type="redo" rotate={-90} /></Button>
        </div>
        {croppedImageUrl && (
          <div className={styles.croppedImageContainer}>
            <img
              alt={formatMessage({id: 'AdminToolsNewsfeedImageCropper.cropAlt'})}
              style={{ maxWidth: '100%' }}
              src={croppedImageUrl}
            />
          </div>
        )}
      </Modal>
    );
  }
}

AdminToolsNewsfeedImageCropper.propTypes = {
  visible: PropTypes.bool,
  handleCancel: PropTypes.func,
  handleOk: PropTypes.func,
  account: PropTypes.object
};
AdminToolsNewsfeedImageCropper.defaultProps = {
  visible: false,
  handleCancel: () => true,
  handleOk: () => true,
  account: {
    firstName: '',
    lastname: '',
    email: '',
    avatar: ''
  }
};
export default AdminToolsNewsfeedImageCropper;
