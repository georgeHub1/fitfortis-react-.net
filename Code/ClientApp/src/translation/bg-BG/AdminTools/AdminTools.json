{
	"header": "За администратори",
	"newsfeed": "Новина",
	"analytics": "Телеметрия"
}
