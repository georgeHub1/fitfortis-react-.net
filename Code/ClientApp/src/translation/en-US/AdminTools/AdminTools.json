{
	"header": "Admin Tools",
	"newsfeed": "Newsfeed",
	"analytics": "Analytics"
}
