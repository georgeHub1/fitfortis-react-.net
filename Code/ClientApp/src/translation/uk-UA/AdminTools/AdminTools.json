{
	"header": "Налаштування",
	"newsfeed": "Новина",
	"analytics": "Аналітика"
}
