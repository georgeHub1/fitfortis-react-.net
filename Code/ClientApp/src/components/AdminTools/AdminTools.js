import React, { Component } from 'react';
import { Tabs } from 'antd';
import { FormattedMessage, injectIntl } from 'react-intl';
import ReactLoading from 'react-loading';
import styles from './styles.module.less';
import CustomCard from '../Common/CustomCard.js';
import AdminToolsEmails from './AdminToolsEmails';
import AdminToolsNewsfeedList from './AdminToolsNewsfeedList';
import AdminToolsAnalytics from './AdminToolsAnalytics';
import { fetchAdminUserListing } from '../../redux/adminTool.action';
import { connect } from 'react-redux';
import { localeSet } from '../../redux/locale.action';
import analyticId from '../../utils/AnalyticsAndAbTests.tsx';


let resizeEvent;

class AdminTools extends Component {
  displayName = AdminTools.name;

  constructor (props) {
    super(props);
    this.formSectionRef = React.createRef();
    this.state = {
      activeTab: 'newsfeed',
      adminUserList: []
    };
  }

  componentDidMount () {
    if (window.location.search.includes('?lang=')) {
      const lan = window.location.search.includes('?lang=') ? window.location.search.substring(6, 11) : (localStorage.getItem('language') || 'en');

      this.props.onUpdateLanguage(lan);
      localStorage.setItem('language', lan);
    }
    if (this.props.userRole === 'user') {
      this.props.history.push('/');
    } else {
      resizeEvent = window.addEventListener('resize', () => {
        const { screenWidth } = this.state;

        if (Math.abs(screenWidth - window.innerWidth) > 100) {
          this.setState({...this.state});
        }
      });
      this.props.fetchAdminUserListing();
    }
    analyticId.firebaseAnalyticsLog('AdminTools_Newsfeed');
  }

  componentWillUnmount () {
    if (resizeEvent) {
      resizeEvent.unsubscribe();
    }
  }

  componentWillReceiveProps (nextProps) {
    if (!this.props.adminTool.isAdminUserListLoaded && nextProps.adminTool.isAdminUserListLoaded) {
      this.setState({...this.state, adminUserList: nextProps.adminTool.adminUserList});
    }
  }

  onChangeTab = activeTab => {
    this.setState({ activeTab });
  };

  render () {
    const { adminUserList, activeTab } = this.state;
    const {
      intl: { formatMessage },
      adminTool: { isAdminUserListFetching },
      account: { email }
    } = this.props;

    return (
      <div className="profilePage">
        <h1 className={styles.header}>
          <FormattedMessage id='AdminTools.header' />
        </h1>
        {
          (!isAdminUserListFetching)
          ? <AdminToolsEmails activeEmail={email} adminUserList={adminUserList}/>
          : <ReactLoading className="react-loading-center" type="spin" color="#335889"  />
        }
        <section className="tabsection">
          <Tabs
            onChange={ this.onChangeTab }
            type="card"
            activeKey={ activeTab }
            tabBarStyle={{ borderBottom: 'none', overflow: 'hidden' }}
          >
            <Tabs.TabPane
              className="ProfileSettings"
              tab={formatMessage({
                id: 'AdminTools.newsfeed'
              })}
              key={ 'newsfeed' }
            >
              <CustomCard
                style={{ borderTopWidth: 0, borderTopRightRadius: 0 }}
              >
                <AdminToolsNewsfeedList />
              </CustomCard>
            </Tabs.TabPane>

            <Tabs.TabPane
              className="ProfileBasicInformation"
              tab={formatMessage({
                id: 'AdminTools.analytics'
              })}
              key={ 'analytics' }
            >
              <CustomCard>
                <AdminToolsAnalytics />
              </CustomCard>
            </Tabs.TabPane>
          </Tabs>
        </section>
      </div>
    );
  }
}

const mapStateToProps = (({ adminTool, profile: { account, userRole } }) => ({ userRole, adminTool, account }));

const mapDispatchToProps = dispatch => ({
  fetchAdminUserListing (...args) {
    return dispatch(fetchAdminUserListing(...args));
  },
  onUpdateLanguage (args) {
    dispatch(localeSet(args));
    return Promise.resolve();
  }
});

const WrappedAdminTools = connect(
  mapStateToProps,
  mapDispatchToProps
)(
  AdminTools
);

export default injectIntl(WrappedAdminTools);
