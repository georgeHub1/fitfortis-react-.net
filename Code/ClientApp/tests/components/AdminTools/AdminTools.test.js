import React from 'react';
import { shallow } from 'enzyme';

import  AdminTools  from '../../../src/components/AdminTools/AdminTools';

const Container = AdminTools;

describe('AdminTools', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Container/>);

    expect(wrapper).toMatchSnapshot();
  });
});
