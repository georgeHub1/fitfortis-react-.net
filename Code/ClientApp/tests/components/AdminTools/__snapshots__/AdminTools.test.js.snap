// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminTools renders correctly 1`] = `ShallowWrapper {}`;
